### DISCUSSION TITLE
LAUNCHXL-CC1352P-4 support

### DISCUSSION THREAD

**Discussion by panciunio** [Koenkk/zigbee2mqtt#5262] [Devices]
Hi, 

It's my first topic raised on zigbee2mqtt discussion board. I decide to put it in this place because it's not an issue neither new feature... I would ask you about one question: Is possible to use LAUNCHXL-CC1352P-4 as zigbee gateway like supported LAUNCHXL-CC1352P-2? I'm willing to use it not only for zigbee but for 433MHz devices in the same time but I'm not sure that firmware CC26X2R1_*.zip will be ok for P4?

Regards,
panciunio

**Comment by Koenkk** [comment]
The P-4 is more optimized towards 433mhz and less for Zigbee (2.4Ghz). Therefore there are no plans to provide a firmware for this device (this is the only thing needed to support it).  You can compile a firmware yourself by looking at the COMPILE.md and firmware.patch from https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0

>  I'm willing to use it not only for zigbee but for 433MHz devices in the same time

I don't know if this is possible at all, Zigbee2MQTT requires a dedicated coordinator.

![image](https://user-images.githubusercontent.com/2892853/101876055-1644cd00-3b8c-11eb-8a0f-c78d7c37b659.png)

**Reply by panciunio** [reply]
Hi @Koenkk,

Thanks for your reply. I was rather curious about the opportunity to use LAUNCHXL-CC1352P-4 as alternative to LAUNCHXL-CC1352P-2 and have chance to test 443 radio for other devices. Normally I'm using  P-2 with external antenna and I'm very satisfied.
Thanks  for your work and good staff!

Regards,
panciunio

**Reply by fredrikgk** [reply]
LAUNCHXL-CC1352P-4 would work fine if the CC1352P FW was rebuilt according to the description here: https://www.ti.com/lit/pdf/swra636

This is because the matching network on the high power PA port on LAUNCXL-CC1352P-4 is designed for 10 dBm operation.
